File: LeetCode/53. Maximum Subarray.swift
class Solution {
func maxSubArray(_ nums: [Int]) -> Int {
    if nums.count > 1 {
        var arr = [Int](repeating: 0, count: nums.count)
        arr[0] = nums[0]
        var max = nums[0]
        for i in 1...nums.count-1 {
            let save = nums[i] + arr[i-1]
            if save > nums[i] {
                arr[i] = save
            }
            else {
                arr[i] = nums[i]
            }
            if arr[i] > max {
                max = arr[i]
            }
        }
        return max
    }
    return nums[0]
}
}

File: LeetCode/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_check = nums[0]
        crnt_sum = nums[0]
        for i in range(1, len(nums)):
            a = crnt_sum + nums[i]
            if a > nums[i]:
                crnt_sum = a
            else:
                crnt_sum = nums[i]
            if crnt_sum > max_check:
                max_check = crnt_sum
        return max_check
